### DISCUSSION TITLE
Library management/improving maintainability

### DISCUSSION THREAD

**Discussion by MrNavaStar** [JetBrains/intellij-micropython#220] [Ideas]
Hi, I am pretty new to micropython, but I really like pycharm, so I want to help contribute to this plugin to help make it more usable. Its not too bad so far, however is a little cumbersome to use at times. Library management is rather annoying as is, so perhaps the plugin should include a way to easily install libraries from the official repo. This could be accomplished using the [mpremote tool](https://docs.micropython.org/en/latest/reference/packages.html).

Another large issue I have come across is that many of the current type hints seem to be outdated. The one I just ran into was machine missing the `TouchPad` class. I think there needs to be a better way of keeping hints up to date. Perhaps hints could be auto generated from the libs repo using a tool like [this](https://pypi.org/project/stub-generator/)? I would be willing to write a program that can do this. Maybe we could setup some github actions that can create new hints for every release of micropython. Then support could be added for choosing a specific version of micropython (running nightlies, older versions, etc)

Auto generating type hints would also help improve the coverage of libraries. I am currently developing on an esp32, and the onewire and ds18x20 libraries are both missing, but they do work if I ignore the errors.

Anyway, I think a discussion needs to be had about improving the library handling and maintainability of this plugin. Let me know what you think!

**Comment by Elizaveta239** [comment]
Hi @MrNavaStar! Thank you very much for the feedback and sorry for the late response! Yes, it's true that currently plugin is in maintenance mode, but pull requests are always welcome!
I'm going to merge #218 soon, so type hints will be improved for ESP32.

If you would like to contribute and submit any other pull requests - that would be awesome, I'm ready to help and answer any questions!
